### PR TITLE
mgr/dashboard: show the host's IP address on Hosts page

### PR DESCRIPTION
**Pull request by aaSharma14**
Fixes: https://tracker.ceph.com/issues/61222

Screenshot from 2023-05-17 19-04-54

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by nizamial09 — APPROVED** [review]

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.html [hunk 1]
```diff
@@ -60,6 +60,16 @@
   </span>
 </ng-template>
 
+<ng-template #hostNameTpl
+             let-row="row">
+  <span [ngClass]="row">
+    {{ row.hostname }}
+  </span><br>
+  <span class="text-muted fst-italic">
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.html:68]: ?
```suggestion
  <span class="text-muted fst-italic"
              *ngIf="row.addr">
```

**Comment by nizamial09** [comment]
@aaSharma14 you'll need to adapt the unit-tests and also run the `npm run fix` locally!
```
 ● HostsComponent › should render hosts list even with not permission mapped services

    expect(received).toBe(expected) // Object.is equality

    Expected: "ceph.dev"
    Received: " ceph.dev "

      115 |       '.datatable-body-cell-label span'
      116 |     );
    > 117 |     expect(spans[0].textContent).toBe(hostname);
          |                                  ^
      118 |   });
      119 |
      120 |   it('should show the exact count of the repeating daemons', () => {

      at src/app/ceph/cluster/hosts/hosts.component.spec.ts:117:34
      at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:409:30)
      at ProxyZoneSpec.Object.<anonymous>.ProxyZoneSpec.onInvoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:3830:43)
      at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:408:56)
      at Zone.Object.<anonymous>.Zone.run (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:169:47)
      at Object.wrappedFunc (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:4330:34)

...............................................................................................................................

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ceph-dashboard@0.0.0 test:ci: `jest --clearCache && JEST_SILENT_REPORTER_DOTS=true jest --coverage --reporters jest-silent-reporter`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ceph-dashboard@0.0.0 test:ci script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jenkins-build/.npm/_logs/2023-05-18T17_45_10_308Z-debug.log
[lint:html    ] 
[lint:html    ] [htmllint] found 0 errors out of 189 files
[lint:html    ] All files pass linting
[lint:prettier] src/app/ceph/cluster/hosts/hosts.component.ts
[lint:eslint  ] 
[lint:eslint  ] Linting "ceph-dashboard"...
[lint:eslint  ] 
[lint:eslint  ] All files pass linting.
[lint:eslint  ] 

Try running 'npm run fix' to fix some linting errors. Some errors might need a manual fix.
```

**Comment by aaSharma14** [comment]
jenkins tets make check

**Comment by aaSharma14** [comment]
jenkins test make check

**Pull request merged**